libsoftwarecontainer/src/softwarecontainer.cpp

```diff
@@ -55,9 +54,10 @@
 
 namespace softwarecontainer {
 
-SoftwareContainer::SoftwareContainer(std::shared_ptr<Workspace> workspace, const std::string &containerID) :
+SoftwareContainer::SoftwareContainer(std::shared_ptr<Workspace> workspace, const ContainerID id) :
```

Comment:
I couldn't reason it, i think agent shouldn't deal with LXC related id. 
The idea of changing id to integer was spin-off from stand-up to get rid of comparision in gateways.